src/watcher.ts

```diff
@@ -0,0 +1,40 @@
+/* eslint-disable @typescript-eslint/naming-convention */
+import { Logger } from '@map-colonies/js-logger';
+import { inject, singleton } from 'tsyringe';
+import { Services } from './common/constants';
+import { IConfigProvider, IFSConfig } from './common/interfaces';
+import { compareDates, getFileContentAsJson } from './common/utils';
+
+@singleton()
+export class Watcher {
+  public constructor(
+    @inject(Services.FSCONFIG) private readonly fsConfig: IFSConfig,
+    @inject(Services.CONFIGPROVIDER) private readonly configProvider: IConfigProvider,
+    @inject(Services.LOGGER) private readonly logger: Logger
+  ) {}
+
+  public async isUpdated(): Promise<boolean> {
+    // gets the last updated time from the saved file
+    const lastUpdatedTimeFromFile = await this.getLastUpdatedTimeFromFile();
```

Comment:
no reason to get it from the file every time, we can read it to local variable.
only if the local variable is empty the we will read from the file